profiler/profile_test.go

```diff
@@ -166,6 +169,51 @@ main.main()
 			"main.indirectShortSleepLoop2",
 		})
 	})
+
+	t.Run("goroutineswait DD_PROFILING_WAIT_PROFILE_MAX_GOROUTINES", func(t *testing.T) {
+		// spawGoroutines spawns n goroutines and then returns a func to stop them.
+		spawnGoroutines := func(n int) func() {
+			launched := make(chan struct{})
+			stopped := make(chan struct{})
+			for i := 0; i < n; i++ {
+				go func() {
+					launched <- struct{}{}
+					stopped <- struct{}{}
+				}()
+				<-launched
```

Comment:
What is the point of `launched`? It looks to me like if you removed it, this would function the same way, since the goroutines would all block on `stopped`.

Reply:
The intend behind `launched` is to ensure that the goroutines have started executing before `spawnGoroutines()` returns. Without `launched` this invariant would be subject to a race condition.

In practice `launched` could be removed without repercussions because the go runtime [immediately registers](https://github.com/golang/go/blob/197a5ee2ab8b64c687c74b986bf92139057366b6/src/runtime/proc.go#L4264-L4385) newly created goroutines as `_Grunnable`. That being said, this behavior seems to be an implementation detail that isn't covered by the language specification. I.e. a future version of Go might decide to put the goroutines into a new `_Glaunchable` state that isn't included in `runtime.NumGoroutines()`. So `launched` could prevent our test from breaking in the future.

That being said, I don't expect the runtime details of this to change anytime soon, so I don't feel strongly about it. I can remove the `launched` if you like.

Reply:
Thanks for the explanation. That makes sense and seems to be a good solution. 

My only comment is that it does not seem obvious why this code is like it is unless you know the go runtime internals. It might be good to make a small comment here about `launched` and how we need the goroutines to be *running* before `spawnGoroutines` returns.

Reply:
Sure, PTAL at 2e9505385671b046eeabd3f308ecdfcb2b1270e3 to see if this does a good enough job in clarifying the intend.